src/main/java/com/photolocos/enterprise/service/PhotoService.java

```diff
@@ -7,53 +7,59 @@
 import org.springframework.stereotype.Component;
 import org.springframework.web.multipart.MultipartFile;
 
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Set;
 
 @Component
 public class PhotoService implements IPhotoService{
+    private IPhotoDAO photoDAO;
+    PhotoDTO photo = new PhotoDTO();
+    Set<PhotoDTO> photos = new HashSet<PhotoDTO>();
 
-    @Autowired
-    IPhotoDAO photoDAO;
-
-
-    @Override
-    public PhotoDTO fetchByLocation(String location) {
-
-        return null;
-    }
-
-    @Override
-    public Set<PhotoDTO> fetchByTag(String tag) {
-        return null;
+    public PhotoService(IPhotoDAO photoDAO) {
+        this.photoDAO = photoDAO;
     }
 
     @Override
     public PhotoDTO savePhoto(PhotoDTO photo, MultipartFile image) {
-        PhotoDTO savedPhoto;
         try {
             photoDAO.save(photo);
             photoDAO.saveImage(image);
-            savedPhoto = photo;
+
+            return photo;
         } catch (Exception e) {
             e.printStackTrace();
-            savedPhoto = new PhotoDTO();
```

Comment:
As long as you're in this catch block, add proper logging.

No one will see printStackTrace() when this is running, and printStackTrace() doesn't have the concept of logging levels, as log4j has.